### PR TITLE
SONAR-7166 fail when projectKey in report is not the component key

### PR DESCRIPTION
**Pull request by sns-seb**
failure implemented as early as possible

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/LoadReportAnalysisMetadataHolderStep.java [hunk 1]
```diff
@@ -39,6 +44,12 @@ public LoadReportAnalysisMetadataHolderStep(BatchReportReader reportReader, Muta
   @Override
   public void execute() {
     BatchReport.Metadata reportMetadata = reportReader.readMetadata();
+    checkState(
+      ceTask.getComponentKey().equals(reportMetadata.getProjectKey()),
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/LoadReportAnalysisMetadataHolderStep.java]: BLOCKER NullPointerException might be thrown as 'ceTask' is nullable here [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/LoadReportAnalysisMetadataHolderStep.java [hunk 2]
```diff
@@ -39,12 +44,31 @@ public LoadReportAnalysisMetadataHolderStep(BatchReportReader reportReader, Muta
...
+  private void checkProjectKeyConsistency(BatchReport.Metadata reportMetadata) {
+    String reportProjectKey = projectKeyFromReport(reportMetadata);
+    checkState(
+      ceTask.getComponentKey().equals(reportProjectKey),
+      "ProjectKey in report (%s) is not consistent with projectKey under which the report as been submitted (%s)",
+      reportProjectKey,
+      ceTask.getComponentKey());
+  }
+
+  private String projectKeyFromReport(BatchReport.Metadata reportMetadata) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/LoadReportAnalysisMetadataHolderStep.java]: MAJOR Make "projectKeyFromReport" a "static" method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/LoadReportAnalysisMetadataHolderStep.java [hunk 3]
```diff
@@ -39,12 +44,31 @@ public LoadReportAnalysisMetadataHolderStep(BatchReportReader reportReader, Muta
...
     mutableAnalysisMetadataHolder.setRootComponentRef(reportMetadata.getRootComponentRef());
     mutableAnalysisMetadataHolder.setBranch(reportMetadata.hasBranch() ? reportMetadata.getBranch() : null);
     mutableAnalysisMetadataHolder.setAnalysisDate(reportMetadata.getAnalysisDate());
     mutableAnalysisMetadataHolder.setCrossProjectDuplicationEnabled(reportMetadata.hasCrossProjectDuplicationActivated() && reportMetadata.getCrossProjectDuplicationActivated());
   }
 
+  private void checkProjectKeyConsistency(BatchReport.Metadata reportMetadata) {
+    String reportProjectKey = projectKeyFromReport(reportMetadata);
+    checkState(
+      ceTask.getComponentKey().equals(reportProjectKey),
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/LoadReportAnalysisMetadataHolderStep.java]: BLOCKER NullPointerException might be thrown as 'ceTask' is nullable here [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* BLOCKER 1 blocker
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by julienlancelot** [comment]
LGTM

**Pull request merged**